query-builder/src/main/java/com/datastax/oss/driver/api/querybuilder/insert/InsertInto.java

```diff
@@ -31,4 +38,29 @@
   /** Makes this statement an INSERT JSON with a bind marker, as in {@code INSERT JSON ?}. */
   @NonNull
   JsonInsert json(@NonNull BindMarker bindMarker);
+
+  /**
+   * Makes this statement an INSERT JSON with a custom type mapping, as in {@code INSERT JSON ?}.
+   *
+   * <p>This is an alternative to {@link #json(String)} for custom type mappings. The provided
+   * registry should contain a codec that can format the value. Typically, this will be your
+   * session's registry, which is accessible via {@code session.getContext().getCodecRegistry()}.
+   *
+   * @throws CodecNotFoundException if {@code codecRegistry} does not contain any codec that can
+   *     handle {@code value}.
+   * @see DriverContext#getCodecRegistry()
+   * @see DefaultInsert#json(Object, CodecRegistry)
+   */
+  @NonNull
+  JsonInsert json(@Nullable Object value, @NonNull CodecRegistry codecRegistry);
```

Comment:
I think this method could have a `default` implementation that calls the `TypeCodec` variant.